app/src/main/java/com/example/nvdovin/weatherapp/factory/GreenDaoFactory.java

```diff
@@ -34,19 +35,26 @@ public void insert(City city) {
         daoSession.getCityDao().insertOrReplace(city);//TODO implement uniqueId logic
     }
 
+    public List<City> loadAllCities(){
+        return daoSession.getCityDao().loadAll();
+    }
+
     public List<City> loadSortedCities(SortQueryBuilder<Property>... queryBuilders) {
```

Comment:
What you method actually does?  You called it **loadSortedCities**, and it takes params,  why do you  load cities 'List<City> cities = loadAllCities();', and then checks args if they aren't null, this is unacceptable and doesn't make sense. Please read carefully about **SOLID** principles.